ethpm/tools/builder.py

```diff
@@ -438,12 +441,228 @@ def validate_link_ref(offset: int, length: int, bytecode: str) -> str:
     return bytecode
 
 
+#
+# Deployments
+#
+
+
+def deployment_type(
+    contract_instance: str,
+    contract_type: str,
+    deployment_bytecode: Dict[str, Any] = None,
+    runtime_bytecode: Dict[str, Any] = None,
+    compiler: Dict[str, Any] = None,
+) -> Manifest:
+    """
+    Returns a `deployment` callable, to which the user *must* include an address and a valid w3
+    instance on which this deployment can be found.
+    """
+    return _deployment_type(
+        contract_instance,
+        contract_type,
+        deployment_bytecode,
+        runtime_bytecode,
+        compiler,
+    )
+
+
+@curry
+def _deployment_type(
+    contract_instance: str,
+    contract_type: str,
+    deployment_bytecode: Dict[str, Any],
+    runtime_bytecode: Dict[str, Any],
+    compiler: Dict[str, Any],
+    address: HexStr,
+    transaction: HexStr = None,
+    block: HexStr = None,
+    w3: Web3 = None,
+) -> Manifest:
+    """
+    Returns a Manifest with the newly included deployment.
+    """
+    validate_address(address)
+    if not w3 and not isinstance(w3, Web3):
+        raise ManifestBuildingError(
+            "A valid w3 instance must be provided to use the deployment_type fn."
+        )
+
+    if not w3.eth.getCode(address):
+        raise ValidationError(f"No code found at the provided address: {address}.")
+        # todo validate code found at address matches runtime_bytecode in contract type.
+
+    if transaction:
+        if not is_hex(transaction):
+            raise ValidationError(f"Transaction: {transaction} is not valid hex.")
+
+        tx_receipt = w3.eth.getTransactionReceipt(transaction)
+        if not tx_receipt:
+            raise ValidationError(f"Tx: {transaction} not found on provided w3.")
+
+        if address and tx_receipt.contractAddress != address:
```

Comment:
do you want to be forgiving here and use `eth_utils.is_same_address`?

Reply:
I also noted that there is a `validate_address` above.  Seems like that validation and this validation should be co-located in the code.